src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -140,6 +142,30 @@ Survey getOne() {
             }
             return surveyMapper.queryPage(DynamoSurvey.class, query).getResults();
         }
+        
+        private List<DynamoSurvey> queryBySurveyIdentifier() {
+            if (studyIdentifier == null) {
+                throw new IllegalStateException("Calculated the need to query by secondary index, but study identifier is not set");
+            }
+            DynamoSurvey hashKey = new DynamoSurvey();
+            hashKey.setStudyIdentifier(studyIdentifier);
+            hashKey.setIdentifier(surveyGuid.substring(11));
```

Comment:
Minor: I believe the withHashKeyValues() only looks at hash key, so setting the identifier won't actually do anything. Since you specify the identifier in the range key, you don't have to specify the identifier here.